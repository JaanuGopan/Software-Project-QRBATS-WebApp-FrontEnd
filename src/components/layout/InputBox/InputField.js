import React from "react";
import "./inputfield.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="input-field"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
