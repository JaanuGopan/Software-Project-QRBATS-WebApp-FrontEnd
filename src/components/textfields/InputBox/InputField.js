import React from "react";
import "../../../pages/Event/EventCreation/EventCreation.css";

const InputField = ({ placeholder, value, onChange, inputType }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className="form-control mb-2"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    </div>
  );
};

export default InputField;
