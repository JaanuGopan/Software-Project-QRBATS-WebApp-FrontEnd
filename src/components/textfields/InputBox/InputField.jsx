import React from 'react';
import '../../../pages/Event/EventCreation/EventCreation.css';

const InputField = ({ placeholder, value, onChange, inputType, error, helperText }) => {
  return (
    <div className="input-with-icon">
      <input
        required
        type={inputType}
        className={`form-control mb-2 ${error ? 'is-invalid' : ''}`}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
      {error && <div className="invalid-feedback">{helperText}</div>}
    </div>
  );
};

export default InputField;
