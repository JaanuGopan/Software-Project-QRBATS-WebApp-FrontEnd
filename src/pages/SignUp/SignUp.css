.signup-container {
    background: url("../../assets/Images/back.jpg") center/cover no-repeat;
    font-family: 'Poppins', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    display: flex;
    justify-content: center;
    align-items: center;
    min-height: 100vh;
    min-width: 300px;
}

.signup-main-container {
    position: relative;
    width: 50%;
    border: 4px solid transparent;
    border-top: 6px solid #0077C0;
    border-radius: 10px;
    padding: 20px;
    background-color: rgba(250, 250, 250, 0.97);
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.1);
    min-width: 250px;
}

.back-button {
    background: #0077C0;
    border: none;
    padding: 10px;
    cursor: pointer;
    outline: none;
    margin-right: 10px;
    border-radius: 50%;
    transition: background-color 0.3s ease;
    display: flex;
    align-items: center;
    justify-content: center;
}

.back-button:hover {
    background: rgba(92, 92, 165, 0.8);
}

.back-button .icon {
    color: #fff;
}

.logo {
    width: 200px;
    height: auto;
    border-radius: 5px;
}

h2 {
    text-align: center;
}

.login-container {
    display: flex;
        flex-direction: row;
        flex-wrap: wrap;
        align-items: center;
        justify-content: center;
}

.image-container {
    flex: 1;
}

.form-container {
    flex: 1;
}

button {
    padding: 10px 20px;
    background-color: #0077C0;
    color: #fff;
    border: none;
    border-radius: 5px;
    cursor: pointer;
}

button:hover {
    background-color: #0056b3;
}

a {
    color: #0077C0;
    text-decoration: none;
}

a:hover {
    background-color: #0056b3;
}

.head2 {
    text-align: center;
    color: #0063A0;
    font-weight: bold;
    font-size: x-large;
    padding: 1%;
    margin: 0%;
}