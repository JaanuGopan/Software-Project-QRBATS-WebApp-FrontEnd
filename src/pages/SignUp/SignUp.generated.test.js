import renderer from 'react-test-renderer';
import React, { useState } from "react";

import axios from "axios";
import logo from "../../assets/Images/signin/Signin.jpeg";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { FaUser } from "react-icons/fa";
import { useNavigate } from "react-router-dom";
import { faCircleArrowLeft } from "@fortawesome/free-solid-svg-icons";
import InputField from "../../components/textfields/InputBox/InputField";
import SignUp from "./SignUp";

jest.mock("./SignUp.css");
jest.mock("axios");
jest.mock("../../assets/Images/signin/Signin.jpeg");
jest.mock("@fortawesome/react-fontawesome");
jest.mock("react-icons/fa");
jest.mock("react-router-dom");
jest.mock("@fortawesome/free-solid-svg-icons");
jest.mock("../../components/textfields/InputBox/InputField");

const renderTree = tree => renderer.create(tree);
describe('<SignUp>', () => {
  it('should render component', () => {
    expect(renderTree(<SignUp 
    />).toJSON()).toMatchSnapshot();
  });
  
});