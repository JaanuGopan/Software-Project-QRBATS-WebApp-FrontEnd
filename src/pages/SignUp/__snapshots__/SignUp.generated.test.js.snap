// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp> should render component 1`] = `
<div
  className="signup-container"
>
  <div
    className="signup-main-container"
  >
    <button
      className="back-button"
      onClick={[Function]}
    />
    <p
      className="head1"
    >
      SIGN UP
    </p>
    <div
      className="login-container"
    >
      <div
        className="image-container"
      >
        <img
          alt="Logo"
          className="logo"
          src="Signin.jpeg"
        />
      </div>
      <div
        className="form-container"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="choice-input mb-3"
          >
             
            <select
              className="select-input"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Select the department
              </option>
              <option
                value="DEIE"
              >
                DEIE
              </option>
              <option
                value="DCOM"
              >
                DCOM
              </option>
              <option
                value="DMME"
              >
                DMME
              </option>
              <option
                value="DCEE"
              >
                DCEE
              </option>
              <option
                value="DMENA"
              >
                DMENA
              </option>
            </select>
          </div>
          <div
            className="choice-input mb-3"
          >
             
            <select
              className="select-input"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Select user role
              </option>
              <option
                value="ADMIN"
              >
                ADMIN
              </option>
              <option
                value="LECTURER"
              >
                LECTURER
              </option>
              <option
                value="STAFF"
              >
                STAFF
              </option>
            </select>
          </div>
          <button
            className="btn btn-primary w-100"
            type="submit"
          >
            Create Account
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
